Mathlib/Analysis/InnerProductSpace/JointEigenspace.lean

```diff
@@ -28,85 +34,108 @@ and a proof obligation that the basis vectors are eigenvectors.
-  exact orthogonalComplement_iSup_eigenspaces_eq_bot <|
-    hB.restrict_invariant <| eigenspace_invariant_of_commute hAB α
+  simpa only [genEigenspace_eq_eigenspace, Submodule.orthogonal_eq_bot_iff]
+    using orthogonalComplement_iSup_eigenspaces_eq_bot <|
+      hB.restrict_invariant <| mapsTo_genEigenspace_of_comm hAB α 1
 
 /-- If A and B are commuting symmetric operators acting on a finite dimensional inner product space,
 then the simultaneous eigenspaces of A and B exhaust the space. -/
-theorem iSup_iSup_eigenspace_inf_eigenspace_eq_top (hA : A.IsSymmetric) (hB : B.IsSymmetric)
-    (hAB : A ∘ₗ B = B ∘ₗ A) :
+theorem iSup_iSup_eigenspace_inf_eigenspace_eq_top_of_commute (hA : A.IsSymmetric)
+    (hB : B.IsSymmetric) (hAB : Commute A B) :
     (⨆ α, ⨆ γ, eigenspace A α ⊓ eigenspace B γ) = ⊤ := by
-  simpa [iSup_eigenspace_inf_eigenspace hB hAB] using
+  simpa [iSup_eigenspace_inf_eigenspace_of_commute hB hAB] using
     Submodule.orthogonal_eq_bot_iff.mp <| hA.orthogonalComplement_iSup_eigenspaces_eq_bot
 
 /-- Given a commuting pair of symmetric linear operators on a finite dimensional inner product
 space, the space decomposes as an internal direct sum of simultaneous eigenspaces of these
 operators. -/
-theorem directSum_isInteral_of_commute (hA : A.IsSymmetric) (hB : B.IsSymmetric)
-    (hAB : A ∘ₗ B = B ∘ₗ A) :
+theorem directSum_isInternal_of_commute (hA : A.IsSymmetric) (hB : B.IsSymmetric)
+    (hAB : Commute A B) :
     DirectSum.IsInternal (fun (i : 𝕜 × 𝕜) ↦ (eigenspace A i.2 ⊓ eigenspace B i.1)):= by
   apply (orthogonalFamily_eigenspace_inf_eigenspace hA hB).isInternal_iff.mpr
   rw [Submodule.orthogonal_eq_bot_iff, iSup_prod, iSup_comm]
-  exact iSup_iSup_eigenspace_inf_eigenspace_eq_top hA hB hAB
-
-end Pair
+  exact iSup_iSup_eigenspace_inf_eigenspace_eq_top_of_commute hA hB hAB
+
+/-- In finite dimensions, the indexed supremum of the joint eigenspaces of a commuting family
+of symmetric linear operators equals `⊤`. -/
+theorem iSup_iInf_eq_top_of_commute {ι : Type*} {T : ι → E →ₗ[𝕜] E}
+    (hT : ∀ i, (T i).IsSymmetric) (h : Pairwise fun i j ↦ Commute (T i) (T j)):
+    ⨆ χ : ι → 𝕜, ⨅ i, eigenspace (T i) (χ i) = ⊤ :=
+  calc
+  _ = ⨆ χ : ι → 𝕜, ⨅ i, maxGenEigenspace (T i) (χ i) :=
+    congr(⨆ χ : ι → 𝕜, ⨅ i,
+      $(maxGenEigenspace_eq_eigenspace (isFinitelySemisimple <| hT _) (χ _))).symm
+  _ = ⊤ :=
+    iSup_iInf_maxGenEigenspace_eq_top_of_iSup_maxGenEigenspace_eq_top_of_commute T h fun _ ↦ by
+    rw [← orthogonal_eq_bot_iff,
+      congr(⨆ μ, $(maxGenEigenspace_eq_eigenspace (isFinitelySemisimple <| hT _) μ)),
+      (hT _).orthogonalComplement_iSup_eigenspaces_eq_bot]
+
+/-- In finite dimensions, given a commuting family of symmetric linear operators, the inner
+product space on which they act decomposes as an internal direct sum of joint eigenspaces. -/
+theorem LinearMap.IsSymmetric.directSum_isInternal_of_commute_of_fintype [Fintype n]
+    (hT : ∀ i, (T i).IsSymmetric) (hC : ∀ i j, Commute (T i) (T j)) :
```

Comment:
`[Fintype n]` is no longer necessary. The switch to `Pairwise` is just to match the previous lemma.
```suggestion
theorem LinearMap.IsSymmetric.directSum_isInternal_of_pairwise_commute [DecidableEq (n → 𝕜)]
    (hT : ∀ i, (T i).IsSymmetric) (hC : Pairwise fun i j ↦ Commute (T i) (T j)) :
```
This should also allow you to replace `fun ⦃_ _⦄ _ ↦ hC _ _` in the proof with `hC`.